### PR TITLE
add define _DEBUG macro in windows env

### PR DESCRIPTION
**Pull request by heheda123123**
image
some lib, such as vld, will judge if this macro is defined.

### CODE REVIEW

File: xmake/rules/mode/xmake.lua [hunk 1]
```diff
@@ -34,6 +34,11 @@ rule("mode.debug")
             if not target:get("optimize") then
                 target:set("optimize", "none")
             end
+            
+            -- enable _DEBUG macros to simulate VS default behaviour
+            if is_plat("windows") then
```
Comment (waruqi) [inline xmake/rules/mode/xmake.lua:39]: target:is_plat

**Comment by waruqi** [comment]
ci 失败了，估计还得切 vs runtime 。。这个改动有点大了。我觉得还是用户自己按需配置比较好，没必要完全跟 vs 的保持一致的。

xmake的原则是默认最小配置

**Comment by heheda123123** [comment]
咦，为啥会失败呢，加这个宏感觉好像没啥影响，或者把vs判断也加上？

**Comment by SirLynix** [comment]
然而，我认为MDd/MD运行时应该被尊重为调试/发布的默认运行时。MT运行时有一些问题（比如你不能从另一个共享库的malloc调用中释放一个指针）。

--

I think however that MDd/MD runtimes should be respected as the default for debug/release. MT runtime has issues (like you cannot free a pointer from a malloc call in another shared library).

**Pull request closed without merging**